tests/contrib/test_gce.py

```diff
@@ -14,102 +14,195 @@
 
 """Unit tests for oauth2client.contrib.gce."""
 
-import json
 from six.moves import http_client
-from six.moves import urllib
 import unittest2
-
+import time
 import mock
+import json
 
 import httplib2
-from oauth2client._helpers import _to_bytes
-from oauth2client.client import AccessTokenRefreshError
-from oauth2client.client import Credentials
-from oauth2client.client import save_to_well_known_file
-from oauth2client.contrib.gce import _DEFAULT_EMAIL_METADATA
-from oauth2client.contrib.gce import _get_service_account_email
-from oauth2client.contrib.gce import _SCOPES_WARNING
-from oauth2client.contrib.gce import AppAssertionCredentials
-
+from oauth2client.client import HttpAccessTokenRefreshError,\
+    Credentials,\
+    save_to_well_known_file
+from oauth2client.contrib.gce import _SCOPES_WARNING,\
+    _METADATA_ROOT,\
+    _get_metadata,\
+    AppAssertionCredentials
+import tempfile
 
 __author__ = 'jcgregorio@google.com (Joe Gregorio)'
 
+METADATA_SERVER = 'oauth2client.contrib.gce._get_metadata'
+DEFAULT_SERVICE_ACCOUNT = {'email': 'default'}
+A_SERVICE_ACCOUNT = {
+    'scopes': [
+        'https://www.googleapis.com/auth/cloud-platform',
+        'https://www.googleapis.com/auth/iam'
+    ],
+    'email': '12345678-compute@developer.gserviceaccount.com'
+}
+
+GET_ACCESS_TOKEN = 'oauth2client.contrib.gce._get_access_token'
+ONE_SECOND_TOKEN = {'access_token': '12345abcde', 'expires_in': 1}
+FOREVER_TOKEN = {'access_token': '12345abcde', 'expires_in': 999999999}
+
 
 class AppAssertionCredentialsTests(unittest2.TestCase):
 
+    # BASIC TESTS
+
     def test_constructor(self):
         credentials = AppAssertionCredentials(foo='bar')
-        self.assertEqual(credentials.scope, '')
         self.assertEqual(credentials.kwargs, {'foo': 'bar'})
         self.assertEqual(credentials.assertion_type, None)
 
     @mock.patch('warnings.warn')
-    def test_constructor_with_scopes(self, warn_mock):
-        scope = 'http://example.com/a http://example.com/b'
-        scopes = scope.split()
-        credentials = AppAssertionCredentials(scope=scopes, foo='bar')
-        self.assertEqual(credentials.scope, scope)
-        self.assertEqual(credentials.kwargs, {'foo': 'bar'})
-        self.assertEqual(credentials.assertion_type, None)
+    @mock.patch(METADATA_SERVER, return_value=A_SERVICE_ACCOUNT)
+    def test_constructor_with_matching_scopes(self, get_metadata, warn_mock):
+        scope = 'https://www.googleapis.com/auth/cloud-platform'
+        AppAssertionCredentials(scope=scope)
```

Comment:
You should actually make assertions about the object created as with done in the deleted `test_constructor_with_scopes`.